### PR TITLE
Add Scene view information text

### PR DESCRIPTION
**Pull request by Kerfuffles**
Adds a small extendable layout group to the lower right of the scene view, displaying brush and model counts. This comes with a preference option to toggle this, which is off by default.

image
image

### CODE REVIEW

File: RealtimeCSG/Assets/Plugins/RealtimeCSG/Editor/Scripts/View/GUI/SceneViewInfoGUI/SceneViewInfo.GUI.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    return;
+
+                GUILayout.BeginArea( infoGUIRect /*, infoGUIBGStyle*/ );
+                {
+                    GUILayout.FlexibleSpace();
+
+                    // -Nuke: internally there is a root model that all other models are grouped under, so we'll subtract 1 to keep an accurate count of user created brushes.
+                    GUILayout.Label
+                    (
+                        ((InternalCSGModelManager.Models.Length - 1 <= 0) ? 0 : InternalCSGModelManager.Models.Length - 1).ToString( "Models:\t###0" ),
```
Comment (Henry00IS) [inline RealtimeCSG/Assets/Plugins/RealtimeCSG/Editor/Scripts/View/GUI/SceneViewInfoGUI/SceneViewInfo.GUI.cs]: Wouldn't it make more sense to use `Math.Max(0, InternalCSGModelManager.Models.Length - 1)` so you only have to fetch the property once? But you'd still end up with two method calls. An even faster approach would be using a local variable on the stack to store the intermediate value and only calculate once:
```C#
int length = InternalCSGModelManager.Models.Length - 1;
((length <= 0) ? 0 : length).ToString("Models:\t###0")
```

File: RealtimeCSG/Assets/Plugins/RealtimeCSG/Editor/Scripts/View/GUI/SceneViewInfoGUI/SceneViewInfo.GUI.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+            try
+            {
+                Handles.BeginGUI();
+
+                if( !CSGSettings.ShowSceneInfo )
+                    return;
+
+                // -Nuke: internally there is a root model that all other models are grouped under, so we'll subtract 1 to keep an accurate count of user created brushes.
+                int modelCount = Mathf.Clamp( InternalCSGModelManager.Models.Length - 1, 0, int.MaxValue );
```
Comment (Henry00IS) [inline RealtimeCSG/Assets/Plugins/RealtimeCSG/Editor/Scripts/View/GUI/SceneViewInfoGUI/SceneViewInfo.GUI.cs]: `int modelCount = Mathf.Max( InternalCSGModelManager.Models.Length - 1, 0 );`

**Comment by LogicalError** [comment]
Hi! I took a look at the PR, nice work
One thing though, which wasn't clear from the images.
I would love it if you added a background behind the text otherwise it'll be hard to see when your scene is a similar color to the text

**Comment by Kerfuffles** [comment]
> I would love it if you added a background behind the text otherwise it'll be hard to see when your scene is a similar color to the text

And done! This should help a lot with the problem described.
image

**Comment by LogicalError** [comment]
Nice 👍

**Pull request merged**